src/django_otp/models.py

```diff
@@ -290,6 +303,68 @@ class VerifyNotAllowed:
     N_FAILED_ATTEMPTS = 'N_FAILED_ATTEMPTS'
 
 
+class GenerationThrottlingMixin(models.Model):
+    """
+    Mixin class for models that need throttling behaviour for generating
+    tokens.
+    """
+
+    last_generated_timestamp = models.DateTimeField(
+        null=True,
+        blank=True,
+        help_text="The last time a token was generated for this device.",
+    )
+
+    def generate_token(self, *args, **kwargs):
+        """
+        Update the timestamp of the last token generation to the current time.
+        """
+        self.last_generated_timestamp = timezone.now()
+        super().generate_token(*args, **kwargs)
```

Comment:
Is `generate_token` guaranteed to exist on all devices? This appears to be a secret dependency on `SideChannelDevice`.

Reply:
This should be resolved now.